django/db/models/base.py

```diff
@@ -1789,6 +1789,8 @@ def _check_fields(cls, **kwargs):
         errors = []
         for field in cls._meta.local_fields:
             errors.extend(field.check(**kwargs))
+        for field in cls._meta.private_fields:
+            errors.extend(field.check(**kwargs))
```

Comment:
One outstanding wrinkle: `private_fields` are [copied into subclasses](https://github.com/django/django/blob/b6f07e2dc69b9a55606f26ac8f13769531148c73/django/db/models/base.py#L347-L350), so checks for one field get repeated in subclasses.

I don’t think there’s a way with the current attributes to tell if a field is local *and* private right now. Maybe more attributes or reorganization could work.